docs/13-configuration.md

```diff
@@ -1,113 +1,128 @@
+
+# wait: <true/false>
+wait: true
+```
+
+Running `aztk spark cluster create` will create a cluster of 4 **Standard\_A2** nodes called 'spark\_cluster' with a linux user named 'spark'. This is equivalent to running the command
+
+```sh
+aztk spark cluster create --id spark --vm-size standard_a2 --size 4 --username spark --wait
+```
+
+NOTE: This assumes that your SSH-key is configured in the *.aztk/secrets.yaml* file.
+
+## *ssh.yaml*
+
+This is the default ssh cluster configuration:
+```yaml
+# username: <name of the user account to ssh into>
+username: spark
+
+# job_ui_port: <local port where the job ui is forwarded to>
+job_ui_port: 4040
+
+# job_history_ui_port: <local port where the job history ui is forwarded to>
+job_history_ui_port: 18080
+
+# web_ui_port: <local port where the spark master web ui is forwarded to>
+web_ui_port: 8080
+
+# jupyter_port: <local port which where jupyter is forwarded to>
+jupyter_port: 8888
+
+# name_node_ui_port: <local port which where Name Node UI is forwarded to>
+name_node_ui_port: 50070
+
+# rstudio_server_port: <local port which where rstudio server is forwarded to>
+rstudio_server_port: 8787
+
+# connect: <true/false, connect to spark master or print connection string (--no-connect)>
+connect: true
+```
+
+Running the command `aztk spark cluster ssh --id <cluster_id>` will ssh into the master node of the Spark cluster. It will also forward the Spark Job UI to localhost:4040, the Spark master's web UI to localhost:8080, and Jupyter to localhost:8888.
+
+Note that all of the settings in ssh.yaml will be overrided by parameters passed on the command line.
+
+## Spark Configuration
+
+The repository comes with default Spark configuration files which provision your Spark cluster just the same as you would locally. After running `aztk spark init` to initialize your working environment, you can view and edit these files at `.aztk/spark-defaults.conf`, `.aztk/spark-env.sh` and `.aztk/core-site.xml`. Please note that you can bring your own Spark configuration files by copying your `spark-defaults.conf`, `spark-env.sh` and `core-site.xml` into your `.aztk/` direcotry.
+
+If using `aztk` job submission, please note that both `spark.shuffle.service.enabled` and `spark.dynamicAllocation.enabled` must be set to true so that the number of executors registered with an application can scale as nodes in the job's cluster come online.
+
+The following settings available in `spark-defaults.conf` and `spark-env.sh` are not supported:
+
+`spark-env.sh`:
+- SPARK\_LOCAL\_IP
+- SPARK\_PUBLIC\_DNS
+- SPARK\_MASTER\_HOST
+- SPARK\_MASTER\_PORT
+- SPARK\_WORKER\_PORT
+- SPARK\_MASTER\_WEBUI\_PORT
+- Any options related to YARN client mode or Mesos
+
+`spark-defaults.conf`:
+- spark.master
+
+Also note that this toolkit pre-loads wasb jars, so loading them elsewhere is not necessary.
+
+### History Server
+If you want to use Spark's history server, please set the following values in your `.aztk/spark-defaults.conf` file:
+```

```

Comment:
Worth linking off to the spark docs for this? (hard part would be to know which version of the docs though...)